File: step_definitions/Verify User Recommendations on Homepage and Product Pages.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import com.page_objects.RecommendationsPage;

public class RecommendationsSteps extends RecommendationsPage {

    private TestHarness testHarness = new TestHarness();

    @Given("a user account with browsing and purchase history")
    public void aUserAccountWithBrowsingAndPurchaseHistory() {
        // Logic to ensure user account has browsing and purchase history
    }

    @When("the user logs into the application")
    public void theUserLogsIntoTheApplication() {
        loginUser();
    }

    @Then("the user is successfully logged in")
    public void theUserIsSuccessfullyLoggedIn() {
        verifyUserLoggedIn();
    }

    @Then("the homepage is displayed with personalized recommendations including electronics and fashion items")
    public void theHomepageIsDisplayedWithPersonalizedRecommendations() {
        verifyHomepageRecommendations();
    }

    @Given("the user is on the homepage")
    public void theUserIsOnTheHomepage() {
        navigateToHomepage();
    }

    @When("the user navigates to a product page of an electronics item")
    public void theUserNavigatesToAProductPageOfAnElectronicsItem() {
        navigateToElectronicsProductPage();
    }

    @Then("the product page displays related accessories recommendations")
    public void theProductPageDisplaysRelatedAccessoriesRecommendations() {
        verifyElectronicsRecommendations();
    }

    @Then("the recommendations are relevant to the electronics item")
    public void theRecommendationsAreRelevantToTheElectronicsItem() {
        verifyRelevanceOfElectronicsRecommendations();
    }

    @When("the user navigates to a product page of a fashion item")
    public void theUserNavigatesToAProductPageOfAFashionItem() {
        navigateToFashionProductPage();
    }

    @Then("the product page displays related fashion accessories recommendations")
    public void theProductPageDisplaysRelatedFashionAccessoriesRecommendations() {
        verifyFashionRecommendations();
    }

    @Then("the recommendations are relevant to the fashion item")
    public void theRecommendationsAreRelevantToTheFashionItem() {
        verifyRelevanceOfFashionRecommendations();
    }

    @Given("the user is on a product page with recommendations")
    public void theUserIsOnAProductPageWithRecommendations() {
        navigateToProductPageWithRecommendations();
    }

    @When("the user adds an item from the recommendations to the cart")
    public void theUserAddsAnItemFromTheRecommendationsToTheCart() {
        addItemFromRecommendationsToCart();
    }

    @Then("the item is successfully added to the cart")
    public void theItemIsSuccessfullyAddedToTheCart() {
        verifyItemAddedToCart();
    }

    @Given("the user has added an item to the cart")
    public void theUserHasAddedAnItemToTheCart() {
        addItemToCart();
    }

    @When("the user checks the recommendations")
    public void theUserChecksTheRecommendations() {
        checkRecommendations();
    }

    @Then("the recommendations adjust to include complementary items")
    public void theRecommendationsAdjustToIncludeComplementaryItems() {
        verifyAdjustedRecommendations();
    }

    @Given("the user logs out of the application")
    public void theUserLogsOutOfTheApplication() {
        logoutUser();
    }

    @When("the user logs back into the application")
    public void theUserLogsBackIntoTheApplication() {
        loginUser();
    }

    @Then("the user is successfully logged in with updated recommendations")
    public void theUserIsSuccessfullyLoggedInWithUpdatedRecommendations() {
        verifyUserLoggedInWithUpdatedRecommendations();
    }

    @Then("the recommendations remain consistent with browsing and purchase behavior")
    public void theRecommendationsRemainConsistentWithBrowsingAndPurchaseBehavior() {
        verifyConsistencyOfRecommendations();
    }

    @Given("the user is logged into the application")
    public void theUserIsLoggedIntoTheApplication() {
        loginUser();
    }

    @When("the user checks recommendations on other sections of the application \(e.g., deals page\)")
    public void theUserChecksRecommendationsOnOtherSectionsOfTheApplication() {
        checkRecommendationsOnOtherSections();
    }

    @Then("the recommendations align with user history across different sections")
    public void theRecommendationsAlignWithUserHistoryAcrossDifferentSections() {
        verifyAlignmentOfRecommendations();
    }

    @Given("the user clears browsing history")
    public void theUserClearsBrowsingHistory() {
        clearBrowsingHistory();
    }

    @Then("the recommendations adjust to focus more on purchase history")
    public void theRecommendationsAdjustToFocusMoreOnPurchaseHistory() {
        verifyRecommendationsFocusOnPurchaseHistory();
    }

    @Given("the user deletes a past purchase from the history")
    public void theUserDeletesAPastPurchaseFromTheHistory() {
        deletePastPurchase();
    }

    @Then("the recommendations adjust, removing items related to the deleted purchase")
    public void theRecommendationsAdjustRemovingItemsRelatedToTheDeletedPurchase() {
        verifyRecommendationsAfterDeletingPurchase();
    }

    @Given("different user profiles with browsing and purchase history")
    public void differentUserProfilesWithBrowsingAndPurchaseHistory() {
        setupDifferentUserProfiles();
    }

    @When("the recommendation verification process is repeated for each user profile")
    public void theRecommendationVerificationProcessIsRepeatedForEachUserProfile() {
        repeatVerificationForUserProfiles();
    }

    @Then("the recommendations are personalized and accurate for each user profile")
    public void theRecommendationsArePersonalizedAndAccurateForEachUserProfile() {
        verifyPersonalizedRecommendationsForProfiles();
    }
}

File: step_definitions/Verify User Recommendations on Homepage and Product Pages.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import RecommendationsPage from "../pages/recommendationsPage";

Given("a user account with browsing and purchase history", () => {
 // Setup user account with history
});

When("the user logs into the application", () => {
 RecommendationsPage.login();
});

Then("the user is successfully logged in", () => {
 RecommendationsPage.verifyLogin();
});

Then("the homepage is displayed with personalized recommendations including electronics and fashion items", () => {
 RecommendationsPage.verifyHomepageRecommendations();
});

Given("the user is on the homepage", () => {
 RecommendationsPage.visitHomepage();
});

When("the user navigates to a product page of an electronics item", () => {
 RecommendationsPage.navigateToElectronicsProductPage();
});

Then("the product page displays related accessories recommendations", () => {
 RecommendationsPage.verifyElectronicsRecommendations();
});

Then("the recommendations are relevant to the electronics item", () => {
 RecommendationsPage.verifyElectronicsRelevance();
});

When("the user navigates to a product page of a fashion item", () => {
 RecommendationsPage.navigateToFashionProductPage();
});

Then("the product page displays related fashion accessories recommendations", () => {
 RecommendationsPage.verifyFashionRecommendations();
});

Then("the recommendations are relevant to the fashion item", () => {
 RecommendationsPage.verifyFashionRelevance();
});

Given("the user is on a product page with recommendations", () => {
 RecommendationsPage.visitProductPageWithRecommendations();
});

When("the user adds an item from the recommendations to the cart", () => {
 RecommendationsPage.addItemToCartFromRecommendations();
});

Then("the item is successfully added to the cart", () => {
 RecommendationsPage.verifyItemAddedToCart();
});

Given("the user has added an item to the cart", () => {
 RecommendationsPage.addItemToCart();
});

When("the user checks the recommendations", () => {
 RecommendationsPage.checkUpdatedRecommendations();
});

Then("the recommendations adjust to include complementary items", () => {
 RecommendationsPage.verifyComplementaryRecommendations();
});

Given("the user logs out of the application", () => {
 RecommendationsPage.logout();
});

When("the user logs back into the application", () => {
 RecommendationsPage.login();
});

Then("the user is successfully logged in with updated recommendations", () => {
 RecommendationsPage.verifyLoginWithUpdatedRecommendations();
});

Then("the recommendations remain consistent with browsing and purchase behavior", () => {
 RecommendationsPage.verifyConsistencyOfRecommendations();
});

Given("the user is logged into the application", () => {
 RecommendationsPage.login();
});

When("the user checks recommendations on other sections of the application (e.g., deals page)", () => {
 RecommendationsPage.checkRecommendationsOnOtherSections();
});

Then("the recommendations align with user history across different sections", () => {
 RecommendationsPage.verifyRecommendationsAcrossSections();
});

Given("the user clears browsing history", () => {
 RecommendationsPage.clearBrowsingHistory();
});

When("the user checks recommendations", () => {
 RecommendationsPage.checkRecommendations();
});

Then("the recommendations adjust to focus more on purchase history", () => {
 RecommendationsPage.verifyRecommendationsFocusOnPurchaseHistory();
});

Given("the user deletes a past purchase from the history", () => {
 RecommendationsPage.deletePastPurchase();
});

When("the user checks recommendations", () => {
 RecommendationsPage.checkRecommendations();
});

Then("the recommendations adjust, removing items related to the deleted purchase", () => {
 RecommendationsPage.verifyRecommendationsAfterDeletingPurchase();
});

Given("different user profiles with browsing and purchase history", () => {
 RecommendationsPage.setupUserProfiles();
});

When("the recommendation verification process is repeated for each user profile", () => {
 RecommendationsPage.repeatVerificationForUserProfiles();
});

Then("the recommendations are personalized and accurate for each user profile", () => {
 RecommendationsPage.verifyPersonalizedRecommendations();
});